### PR TITLE
BuildResidentialHPXML: Allow separate furnace as ductless heat pump backup

### LINKED ISSUE #1393: BuildResidentialHPXML: Allow "second" furnace
As long as first system is ductless.

(Same for fixed heater?)

See https://github.com/NREL/resstock/pull/1074#discussion_r1195546890 for context.

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Closes #1393.

## Checklist

PR Author: Check these when they're done. Not all may apply. ~~strikethrough~~ and check any that do not apply. 

PR Reviewer: Verify each has been completed.

- [ ] Schematron validator (`EPvalidator.xml`) has been updated
- [ ] Sample files have been added/updated (via `tasks.rb`)
- [ ] Tests have been added/updated (e.g., `HPXMLtoOpenStudio/tests` and/or `workflow/tests/hpxml_translator_test.rb`)
- [ ] Documentation has been updated
- [ ] Changelog has been updated
- [ ] `openstudio tasks.rb update_measures` has been run
- [ ] No unexpected changes to simulation results of sample files

### CODE REVIEW

File: BuildResidentialHPXML/measure.rb [hunk 1]
```diff
@@ -3214,6 +3216,12 @@ def argument_errors(args)
     error = (args[:heating_system_type] == 'none') && (args[:heat_pump_type] == 'none') && (args[:heating_system_2_type] != 'none')
     errors << 'A second heating system was specified without a primary heating system.' if error
 
+    if (args[:heat_pump_backup_type] == HPXML::HeatPumpBackupTypeSeparate) && [HPXML::HVACTypeFurnace, HPXML::HVACTypeFixedHeater].include?(args[:heating_system_2_type]) # separate ducted backup
```
Comment (shorowit) [inline BuildResidentialHPXML/measure.rb:3219]: A fixed heater is not ducted, it's the same as a portable heater except can't be moved around.

File: BuildResidentialHPXML/tests/build_residential_hpxml_test.rb [hunk 2]
```diff
@@ -210,6 +211,7 @@ def test_workflows
       'error-ducts-location-and-areas-not-same-type.xml' => 'Duct location and surface area not both defaulted or not both specified.',
       'error-second-heating-system-serves-total-heat-load.xml' => 'The fraction of heat load served by the second heating system is 100%.',
       'error-second-heating-system-but-no-primary-heating.xml' => 'A second heating system was specified without a primary heating system.',
+      'error-second-heating-system-ducted-with-ducted-primary-heating.xml' => 'A ducted heat pump with "separate" ducted backup is not supported.',
```
Comment (shorowit) [inline BuildResidentialHPXML/tests/build_residential_hpxml_test.rb:214]: 👍

**Review by shorowit — APPROVED** [review]
Looks good to me. Thanks @joseph-robertson. I'll merge when the CI comes back green after my last minor commit.

**Comment by joseph-robertson** [comment]
The PR title doesn't reflect this, but this also allows fixed heater as a valid second heating system type.

**Pull request merged**